### DISCUSSION TITLE
Iceoryx hoofs for functional safety

### DISCUSSION THREAD

**Discussion by hemalbavishi** [eclipse-iceoryx/iceoryx#1386] [Q&A] [answered]
Hello,

I am just wondering whether unix domain socket functionalities within Iceoryx hoofs are compliant to functional safety.

https://github.com/eclipse-iceoryx/iceoryx/blob/master/iceoryx_hoofs/source/posix_wrapper/unix_domain_socket.cpp

This question is arise because it is using std::string instead of cxx::string. Any reason why std::string is used instead of cxx::string

**Comment by elBoberido** [comment, marked as answer]
@hemalbavishi this is one of the classes which needs some refactoring before it is considered for use in a safety environment. This is part of #260 and we want to have this cleanups done by the release of iceoryx v3.0

**Reply by hemalbavishi** [reply]
@elBoberido : Thank you for your quick response. Any idea when Iceoryx v3.0 will be released?

**Reply by mossmaurice** [reply]
@hemalbavishi Currently, the C release `v3.0` is roughly planned for end of 2022 or beginning 2023.
